### PR TITLE
[IOTDB-4301] [IOTDB-4303] fix bug in create regions group procedure

### PR DESCRIPTION
**Pull request by Beyyes**
## Description

see details in the jira: 

https://issues.apache.org/jira/browse/IOTDB-4303?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=17599234#comment-17599234,

https://issues.apache.org/jira/browse/IOTDB-4301?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel

### CODE REVIEW

**Comment by MiniSho** [comment]
add IOTDB-4232 to the title, it's the same bug.
LGTM~

File: confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/CreateRegionGroupsProcedure.java [hunk 1]
```diff
@@ -130,6 +153,52 @@ protected int getStateId(CreateRegionGroupsState createRegionGroupsState) {
...
+  @Override
+  public boolean equals(Object that) {
+    if (that instanceof CreateRegionGroupsProcedure) {
+      CreateRegionGroupsProcedure thatProc = (CreateRegionGroupsProcedure) that;
+      return thatProc.getProcId() == this.getProcId()
+          && thatProc.getState() == this.getState()
+          && thatProc.createRegionGroupsPlan.equals(this.createRegionGroupsPlan)
+          && thatProc.failedRegions.equals(this.failedRegions);
+    }
+    return false;
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/CreateRegionGroupsProcedure.java:202]: implementation  @hashcode

**Review by CRZbulabula — APPROVED** [review]
LGTM～

File: confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/CreateRegionGroupsProcedure.java (hunk 1 above)
Comment (Beyyes) [reply confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/CreateRegionGroupsProcedure.java:203]: fix

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**